Create rule S6509: Extra boolean casts should be removed (#1619)
SonarSource/rspec: rules/S6509/javascript/rule.adoc

> In JavaScript, every value has an associated boolean value: either ``true`` or ``false``.

Reviewer: ```suggestion
In JavaScript, every value can be coerced into a boolean value: either ``true`` or ``false``.
```

Reviewer 2: 👍

> Values that are associated with ``true`` are said to be _truthy_, and those associated with ``false`` are said to be _falsy_.

Reviewer: ```suggestion
Values that are coerced into ``true`` are said to be _truthy_, and those coerced into ``false`` are said to be _falsy_.
```
